client/js/screens/SignIn/SignIn.js

```diff
@@ -15,136 +19,192 @@ import InputDefaultField from '../../components/InputDefaultField';
 import {THEME} from '../../config';
 import styles from './styles';
+import {addViewer} from '../../config/models';
+import PropTypes from 'prop-types';
 
-const SignIn = props => {
+const client = new ApolloClient({uri: 'http://157.245.163.7:8000/'});
+
+const SignIn = ({navigation}) => {
   const [isErrors, setErrors] = useState(null);
+
   return (
-    <>
-      <ImageBackground
-        style={styles.signInBgImg}
-        source={require('../../assets/images/imgSignIn01.jpg')}
-      />
-      <ScrollView>
-        <SafeAreaView style={THEME.padding.accountScreen}>
-          <TitleText style={styles.title}>Sign In</TitleText>
-          <Form onSubmit={values => {}}>
-            {props => {
-              return (
-                <View>
-                  <FormSpy
-                    subscription={{values: true}}
-                    onChange={({values}) => {
-                      setErrors(null);
-                      if (!values.email) {
-                        setErrors({
-                          ...isErrors,
-                          email: 'Please enter the email',
-                        });
-                      }
-                      if (!values.password) {
-                        setErrors({
-                          ...isErrors,
-                          password: 'Please enter the password',
-                        });
-                      }
-                    }}
-                  />
-                  <Field name="email" placeholder="example@email.com">
-                    {({input, meta, placeholder}) => {
-                      return (
-                        <InputDefaultField
-                          {...input}
-                          title="Email"
-                          autoCapitalize="none"
-                          autoCompleteType="email"
-                          placeholder={placeholder}
-                          textContentType="emailAddress"
-                          style={styles.inputEmail}
-                        />
+    <Mutation
+      mutation={MUTATION_LOG_IN}
+      client={client}
+      onCompleted={async ({login}) => {
+        const viewer = {id: login.user.id, token: login.token};
+        await addViewer(viewer);
+        await navigation.navigate('AuthLoading');
+        try {
+        } catch (error) {}
+      }}>
+      {(logIn, {loading, error, data}) => {
+        return (
+          <>
+            <ImageBackground
+              style={styles.signInBgImg}
+              source={require('../../assets/images/imgSignIn01.jpg')}
+            />
+            <ScrollView>
+              <SafeAreaView style={THEME.padding.accountScreen}>
+                <TitleText style={styles.title}>Sign In</TitleText>
+                <Form
+                  onSubmit={async values => {
+                    try {
+                      await logIn({variables: values});
+                    } catch (error) {
+                      let message = error.message.split(': ')[1];
```

Comment:
this code itself can also throw errors, if error.message is undefined, or the split doesn't return an array. you should write more checks (i.e msg = error ? error.message ? X : "generic error")